### DISCUSSION TITLE
Milvus and Cache database

### DISCUSSION THREAD

**Discussion by dod-ia** [milvus-io/milvus#17070] [Q&A and General discussion]
Hello,

Is Milvus designed to work as a Cache Database (like Redis for example) in addition to the vector searching ?
For example, when the "search by id" feature will be done, using Milvus as a key:value database can be a correct approach or Milvus is not designed to do that ?
The Use Case may be a search by id first and, if no match, realizing a vector search.

Thanks,

**Comment by yhmo** [comment]
In Milvus, the "search by id" is a two-steps search operation:
1. retrieve a vector by an id
2. use the vector to do ANN search within the database

From your use case description, I guess your requirement is:
1. you have an id and a vector, firstly you want to test whether this id is already exist in Milvus
2. if the id already exist in Milvus, do nothing
3. else, use the vector to do search

To test whether an id is exist in Milvus, you can use query() interface. Before you call this interface, Milvus requires to load the entire collection into CPU memory. If the query() return empty result, call search() to do ANN search.

**Reply by dod-ia** [reply]
Thanks for your response !  The query search is comparable in performance to a cache database request ? In my use-case, I have to realize the id checking step lots of times and it can be a bottleneck if the query velocity is not quick enough.

**Reply by xiaofan-luan** [reply]
> 

Hi moutroll, we don't think milvus has comparable performance with redis when  query by id, we just leave it there to save your save your effort.
How ever, we'd like to improve the query speed and we already did something. It should be different in 2.1~
